Core/PersistentPixelStoring.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  PersistentPixelStoring.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Networking
+
+public struct PersistentPixelMetadata: Identifiable, Codable, Equatable {
+
+    public enum PixelType: Codable {
+        case daily
+        case count
+        case regular
+    }
+
+    public let id: UUID
+    public let eventName: String
+    public let pixelType: PixelType
+    public let additionalParameters: [String: String]
+    public let includedParameters: [Pixel.QueryParameters]
+
+    public init(eventName: String,
+                pixelType: PersistentPixelMetadata.PixelType,
+                additionalParameters: [String: String],
+                includedParameters: [Pixel.QueryParameters]) {
+        self.id = UUID()
+        self.eventName = eventName
+        self.pixelType = pixelType
+        self.additionalParameters = additionalParameters
+        self.includedParameters = includedParameters
+    }
+
+    var pixelName: String {
+        switch pixelType {
+        case .daily: return eventName + "_d"
+        case .count: return eventName + "_c"
+        case .regular: return eventName
+        }
+    }
+
+    var timestamp: String? {
+        return additionalParameters[PixelParameters.originalPixelTimestamp]
+    }
+}
+
+protocol PersistentPixelStoring {
+    func append(pixels: [PersistentPixelMetadata]) throws
+    func remove(pixelsWithIDs: Set<UUID>) throws
+    func storedPixels() throws -> [PersistentPixelMetadata]
+}
```

Comment:
There's now no way to remove pixels from the array without using them by ID, so we don't have to worry about newly-appended pixels getting overwritten somehow, like was possible with the old approach.